src/directives/sortable.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @fileoverview Provides a directive that can be used to change the order of
+ * items in an array.
+ * It can be used to change the order of layers in the map for example.
+ * It requires JQuery-UI Sortable.
+ *
+ * Example:
+ *
+ * <ul ngeo-sortable="ctrl.map.getLayers().getArray()">
+ *   <li ng-repeat="ctrl.map.getLayers().getArray()">
+ *      <span class="handle"></span>{{layer.get('name')}}
+ *   </li>
+ * </ul>
+ */
+
+goog.provide('ngeo.sortableDirective');
+
+goog.require('ngeo');
+
+
+/**
+ * @param {angular.$log} $log Angular log service.
+ * @param {angular.$timeout} $timeout Angular timeout service.
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+ngeo.sortableDirective = function($log, $timeout) {
+  return {
+    restrict: 'A',
+    scope: true,
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Attributes.
+         */
+        function(scope, element, attrs) {
+          var attr = 'ngeoSortable';
+          var prop = attrs[attr];
+
+          var sortable = /** @type {Array} */ (scope.$eval(prop));
+
+          if (!angular.element.fn || !angular.element.fn.jquery) {
+            $log.error('ui.sortable: jQuery should be included before' +
+                'AngularJS!');
+            return;
+          }
+
+          var startIndex;
+
+          // When we add or remove elements, we need the sortable to 'refresh'
+          // so it can find the new/removed elements.
+          scope.$watch('sortable.length', function() {
+            // Timeout to let ng-repeat modify the DOM
+            $timeout(function() {
+              element.sortable('refresh');
+            }, 0, false);
+          });
+
+
+          var callbacks = {};
+          callbacks.start = function(e, ui) {
+            // Save the starting position of dragged item
+            startIndex = ui.item.index();
+          };
+
+          callbacks.update = function(e, ui) {
+            scope.$apply(function() {
+              sortable.splice(
+                  ui.item.index(), 0,
+                  sortable.splice(startIndex, 1)[0]);
+            });
+            startIndex = undefined;
+          };
+
+
+          element.sortable();
+
+          angular.forEach(callbacks, function(value, key) {
+            element.sortable('option', key, value);
+          });
```

Comment:
This won't work if the `callbacks` object's `start` and `update` keys are renamed. It's just two options, so I don't think we need to use a loop.